src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Syntax/SyntaxListOfT.cs

```diff
@@ -546,7 +502,7 @@ public TNode Current
         {
             get
             {
-                return (TNode)_list.ItemInternal(_index);
+                return (TNode)_list.ItemInternal(_index)!;
```

Comment:
Maybe we should use `AssumeNotNull` in places like this to fail early. Or some equivalent helper that only does `Debug.Assert`.

Reply:
Yeah, I can change that to `AssumeNotNull`. In this case, I was just following the Roslyn annotations for the same type.